src/Components/Components/src/Lifetime/IComponentApplicationStateStore.cs

```diff
@@ -0,0 +1,24 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// Manages the storage for components and services that are part of a Blazor application.
+    /// </summary>
+    public interface IComponentApplicationStateStore
+    {
+        /// <summary>
+        /// Gets the persisted state from the store.
+        /// </summary>
+        /// <returns>The persisted state.</returns>
+        IDictionary<string, byte[]> GetPersistedState();
```

Comment:
What's the reason for this being synchronous-only?

I know the components themselves need to be able to retrieve key-value pairs synchronously to avoid extra rendering, but I would have thought the application startup process could wait for an async loading process if it wanted.

Or is the reasoning that nobody apart from us will ever implement this interface, and we will never want to support anything that loads asynchronously? Conceptually, I could imagine some future storage that's not in the HTML file but rather goes into an external Redis store or similar.

Reply:
This was an oversight on my part, it should be async